ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py

```diff
@@ -3917,3 +3917,38 @@ def test_tensorflow_Zeta(
         x=x[0],
         q=x[1],
     )
+
+
+# Imag
+@handle_frontend_test(
+    fn_tree="tensorflow.raw_ops.Imag",
+    dtype_and_xs=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("complex"),
+    ),
+    Tout=st.sampled_from(["float64", "float32"]),
+    test_with_out=st.just(False),
+)
+def test_tensorflow_Imag(  # NOQA
+    *,
+    dtype_and_xs,
+    Tout,
+    frontend,
+    test_flags,
+    fn_tree,
+    on_device,
+):
+    input_dtype, xs = dtype_and_xs
+    if input_dtype[0] == "complex128":
+        Tout = "float64"
+    elif input_dtype[0] == "complex64":
+        Tout = "float32" if Tout else None
```

Comment:
A slight renaming would be better. Because it's counter-intuitive to use `Tout` for 2 different uses. Maybe the test parameter one should be named `send_type` or something similar.